onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -379,45 +386,56 @@ void TreeEnsembleCommon<InputType, ThresholdType, OutputType>::ComputeAgg(concur
             [this, &scores, &agg, x_data](ptrdiff_t j) {
               agg.ProcessTreeNodePrediction1(scores[j], *ProcessTreeNodeLeave(roots_[j], x_data));
             },
-            0);
+            max_num_threads);
 
         for (auto it = scores.cbegin(); it != scores.cend(); ++it) {
           agg.MergePrediction1(score, *it);
         }
       }
       agg.FinalizeScores1(z_data, score, label_data);
-    } else if (N <= parallel_N_) { /* section C: 1 output, 2+ rows but not enough rows to parallelize */
-      ScoreValue<ThresholdType> score;
+    } else if (N <= parallel_N_ || max_num_threads == 1) { /* section C: 1 output, 2+ rows but not enough rows to parallelize */
+      std::vector<ScoreValue<ThresholdType>> scores(parallel_tree_N_);
       size_t j;
+      int64_t i, batch, batch_end;
 
-      for (int64_t i = 0; i < N; ++i) {
-        score = {0, 0};
+      for (batch = 0; batch < N; batch += parallel_tree_N_) {
```

Comment:
Would be good to capture in a comment why we're splitting things into batches even though we're not parallelizing here, as it's not clear how that provides a benefit.

Reply:
I added one comment to explain why it is faster to do it that way.